src/diagnostics/writers.jl

```diff
@@ -0,0 +1,160 @@
+# Writers.jl
+#
+# This file defines function-generating functions for output_writers for diagnostics. The
+# writers come with opinionated defaults.
+
+import ClimaCore.Remapping: interpolate_array
+import NCDatasets
+
+"""
+    HDF5Writer()
+
+
+Save a `ScheduledDiagnostic` to a HDF5 file inside the `output_dir` of the simulation.
+
+
+TODO: This is a very barebone HDF5Writer. Do not consider this implementation as the "final
+word".
+
+We need to implement the following features/options:
+- Toggle for write new files/append
+- Checks for existing files
+- Check for new subfolders that have to be created
+- More meaningful naming conventions (keeping in mind that we can have multiple variables
+  with different reductions)
+- All variables in one file/each variable in its own file
+- All timesteps in one file/each timestep in its own file
+- Writing the correct attributes
+- Overriding simulation.output_dir (e.g., if the path starts with /)
+- ...more features/options
+
+"""
+function HDF5Writer()
+    # output_drivers are called with the three arguments: the value, the ScheduledDiagnostic,
+    # and the integrator
+    function write_to_hdf5(value, diagnostic, integrator)
+        var = diagnostic.variable
+        time = integrator.t
+
+        # diagnostic here is a ScheduledDiagnosticIteration. If we want to obtain a
+        # descriptive name (e.g., something with "daily"), we have to pass the timestep as
+        # well
+
+        output_path = joinpath(
+            integrator.p.simulation.output_dir,
+            "$(diagnostic.output_name)_$time.h5",
```

Comment:
I think by default we should write everything at the same output interval to one file.